test/e2e/e2e_tests_suite_test.go

```diff
@@ -0,0 +1,221 @@
+		if err != nil {
+			return fmt.Errorf("failed to create debug pod on node %s: %w", node.Name, err)
+		}
+
+		defer func() {
+			err := utilsHelpers.DeletePod(debugPod)
+			if err != nil {
+				fmt.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+			}
+		}()
+
+		command := []string{
+			"bash", "-c",
+			fmt.Sprintf("mkdir -p nft && echo '%s' > nft/firewall.nft", string(NFTtable)),
+		}
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to save rule set to file on node %s: %w", node.Name, err)
+		}
+
+		installCmd := []string{"bash", "-c", "yum update -y && yum install -y nftables"}
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, installCmd)
+		if err != nil {
+			return fmt.Errorf("failed to save rule set to file on node %s: %w", node.Name, err)
+		}
+
+		command = []string{
+			"bash", "-c", "/usr/sbin/nft -f nft/firewall.nft",
+		}
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to apply rule set %q on node %s: %w", command, node.Name, err)
+		}
+
+		// Check the output of nft list ruleset
+		command = []string{"bash", "-c", "/usr/sbin/nft list ruleset"}
+		output, err := utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to list NFT ruleset on node %s: %w", node.Name, err)
+		}
+		fmt.Printf("Command output: %s\n", string(output))
+
+		if len(output) == 0 {
+			return fmt.Errorf("no output from 'nft list ruleset' on node %s: ", node.Name)
+		}
+		command = []string{
+			"bash", "-c",
+			"/usr/sbin/nft list ruleset > /etc/nftables.conf",
+		}
+
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to save NFT ruleset to file on node %s: %w", node.Name, err)
+		}
+
+		command = []string{"bash", "-c",
+			"systemctl enable nftables",
+		}
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to enable nftables %q on node %s: %w", command, node.Name, err)
+		}
+
+		artifactsDir, _ := os.LookupEnv("ARTIFACT_DIR")
+		comArtifactsDir := filepath.Join(artifactsDir, "commatrix")
+		err = os.MkdirAll(comArtifactsDir, 0755)
+		if err != nil {
+			return fmt.Errorf("failed to create artifactsDir %w", err)
+		}
+
+		// save the nft ruleset on ArtifactDirPath.
+		err = os.WriteFile(filepath.Join(comArtifactsDir, fmt.Sprintf("nftables-%s.nft", node.Name)), output, 0644)
+		if err != nil {
+			return fmt.Errorf("failed to save nft file on for node artifacts %s %w", node.Name, err)
+		}
+	}
+	return nil
+}
+
+// isSNOCluster will check if OCP is a single node cluster.
+func isSNOCluster(cs *client.ClientSet) (bool, error) {
+	nodes, err := cs.CoreV1Interface.Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return false, err
+	}
+	if len(nodes.Items) > 1 {
```

Comment:
Finding one node can also mean the cluster is unhealthy (MNO with broken node), better check the cluster infrastructure status